src/views/preview/preview.jsx

```diff
@@ -196,6 +211,21 @@ class Preview extends React.Component {
             );
         }
     }
+    handleToggleStudio (event) {
+        const studioId = parseInt(event.currentTarget.dataset.id, 10);
+        if (!isNaN(studioId)) { // probably unnecessary test; should always be a number
```

Comment:
I agree that you don't need this test, since you check if it's present with `this.props.studios.find` — non-existent studio ids including `NaN` will be handled by that.

Reply:
<programmer opinion>I think it's bad form to pass on a state that indicates an error on to subsequent code, just because that code happens to handle the error case. What about when others change this in the future?</programmer opinion>